src/Compilers/Core/Portable/CodeGen/ITokenDeferral.cs

```diff
@@ -13,7 +13,12 @@ internal interface ITokenDeferral
         uint GetFakeSymbolTokenForIL(Cci.ISignature value, SyntaxNode syntaxNode, DiagnosticBag diagnostics);
         uint GetSourceDocumentIndexForIL(Cci.DebugSourceDocument document);
 
-        Cci.IFieldReference GetFieldForData(ImmutableArray<byte> data, SyntaxNode syntaxNode, DiagnosticBag diagnostics);
+        Cci.IFieldReference GetFieldForData(ImmutableArray<byte> data, ushort alignment, SyntaxNode syntaxNode, DiagnosticBag diagnostics);
+
+        /// <summary>Gets a field that may be used to lazily cache an array created to store the specified data.</summary>
+        /// <remarks>This is used to cache an array created with the data passed to <see cref="GetFieldForData"/>.</remarks>
+        Cci.IFieldReference GetArrayCachingFieldForData(ImmutableArray<byte> data, Cci.IArrayTypeReference arrayType, SpecialType elementType, SyntaxNode syntaxNode, DiagnosticBag diagnostics);
```

Comment:
>SpecialType elementType [](http://example.com/codeflow?start=118&length=23)

Can't we get this from ```arrayType```?

Reply:
I wanted to do that but couldn't figure out how.  There's an IArrayTypeReference.GetElementType method but it requires an EmitContext instance to be passed in, and I wasn't sure where to get that from.  If you know, I'd be happy to change it.

Reply:
>I wanted to do that but couldn't figure out how.  There's an IArrayTypeReference.GetElementType method but it requires an EmitContext instance to be passed in, and I wasn't sure where to get that from.  If you know, I'd be happy to change it.

```CommonPEModuleBuilder``` creates one on the fly in several places.

Reply:
The two such occurrences I saw do:
```C#
new EmitContext(this, syntaxNode, diagnostics, metadataOnly: false, includePrivateMembers: true)
```
If it's valid here to similarly assume metadataOnly=false and includePrivateMembers=true (I don't know what the implications of that are), then I should be able to take this approach if it's considered better than just passing in the SpecialType we already have.

Reply:
I think ```includePrivateMembers=false``` is meaningful on;y when ```metadataOnly=true```., And since we are emitting IL (we wouldn't need to call the helpers otherwise, I think), ```metadataOnly=false``` feels appropriate

Reply:
> if it's considered better than just passing in the SpecialType we already have.

We should avoid passing redundant data around. Also, we should avoid creating a situation where one can pass inconsistent input by mistake. I.e. pass ```SpecialType``` that doesn't match array's element type, for example.